### ISSUE TITLE
WSL2 Experimental Features Break Rancher Desktop

### ISSUE THREAD

**Issue by flockoftanks** [rancher-sandbox/rancher-desktop#5607]
### Actual Behavior

After enabling WSL2 version 2.0.0, I used the available experimental features to test them. Those features are documented here [GAMINGDEPUTY](https://www.gamingdeputy.com/wsl-2-0-update-2-0-2-windows-subsystem-for-linux-with-new-experimental-features-and-more/), so I won't bother getting into them more.

Immediately after enabling these, I noticed that Rancher Desktop could no longer fully initialize and Docker ceased being able to run containers in WSL2. Rancher desktop would start up and get stuck on Initializing Rancher Desktop. It would sit in that state for hours with no error displayed.

### Steps to Reproduce

1. Install WSL2 2.0.0
2. enable all experimental features
3. start Rancher Desktop.
4. make sure Docker (Moby) container runtime is selected
5. watch it fail to start up

### Result

No errors were displayed, but it would sit idle on "Initializing Rancher Desktop" and never timeout.

### Expected Behavior

Rancher Desktop would run fully and docker images could be run in WSL2 via the WSL integration.

This issue happens with or without kubernetes enabled.

### Additional Information

I enabled the experimental features one-by-one and was able to narrow the problem down to the property `autoMemoryRelcaim=gradual`.  I don't know if the other settings for `automaticMemoryRelcaim` are also broken. I assume this feature somehow stops the docker engine from running because I couldn't run containers on the host OS either.

### Rancher Desktop Version

1.10.0

### Rancher Desktop K8s Version

1.28.2

### Which container engine are you using?

moby (docker cli)

### What operating system are you using?

Windows

### Operating System / Build Version

Windows 11 Pro 23H2

### What CPU architecture are you using?

x64

### Linux only: what package format did you use to install Rancher Desktop?

None

### Windows User Only

Company VPN, but this issue occurred even when disconnected from install -> Rancher Desktop startup.

**Comment by asos-chrishimsworth** [comment]
I had the same issue and thanks to your issue pointing me in the right direction, managed to fix it.  If you update the `.wslconfig` file (in `C:\Users\<yourusername>\.wslconfig`) to the following, then Rancher Desktop should start:
```
[experimental]
autoMemoryReclaim=disabled
```
More details [here](https://devblogs.microsoft.com/commandline/windows-subsystem-for-linux-september-2023-update/)

**Comment by eoptam** [comment]
The `dropcache` setting for `autoMemoryReclaim` seems to work for me with Rancher Desktop v1.10.0:
```
[experimental]
autoMemoryReclaim=dropcache
```

**Comment by jandubois** [comment]
> The `dropcache` setting for `autoMemoryReclaim` seems to work for me with Rancher Desktop v1.10.0:

Yes, it is the "gradual" setting that is incompatible with Rancher Desktop because it disables cgroup v1.

**Comment by mook-as** [comment]
As of Rancher Desktop 1.15, we only use cgroup v2 (because we dropped support for older Kubernetes versions that required cgroup v1).  This was done as part of #7180.

This means that `autoMemoryReclaim=gradual` now works correctly.  I'm using this as a test configuration:
```
pageReporting=true
networkingMode=mirrored
dnsTunneling=true
autoProxy=true
autoMemoryReclaim=gradual
sparseVhd=true
useWindowsDnsCache=true
bestEffortDnsParsing=true
```

I also tried running `nginx`, exposing it on port 5000, and connecting to it with all network interfaces.
